electionleaflets/apps/leaflets/views.py

```diff
@@ -89,6 +93,28 @@ def get_context_data(self, **kwargs):
             self.object,
             self.request.user
         )
+
+        if self.object.ynr_person_id and self.object.ynr_person_name:
+            context['person_link'] = 'https://candidates.democracyclub.org.uk/person/{}'.format(self.object.ynr_person_id)
+            context['person_name'] = self.object.ynr_person_name
+        elif self.object.publisher_person:
+            context['person_link'] = reverse('person', kwargs={'remote_id': self.object.publisher_person.remote_id})
+            context['person_name'] = self.object.publisher_person.name
+
+        if self.object.ynr_party_id and self.object.ynr_party_name:
+            pp_id = 'PP{}'.format(re.sub(r'party:', '', self.object.ynr_party_id))
+            context['party_link'] = reverse('party-view', kwargs={'pk': pp_id})
+            context['party_name'] = self.object.ynr_party_name
+        elif self.object.publisher_party:
+            context['party_link'] = reverse('party-view', kwargs={'pk': self.object.publisher_party.pk})
+            context['party_name'] = self.object.publisher_party.party_name
+
+        if self.object.ballot_id:
+            r = devs_dc_helper.ballot_request(self.object.ballot_id)
+            if r.status_code == 200:
+                ballot = r.json()
+                context['ballot_name'] = '{} in {}'.format(ballot['election_title'], ballot['election_type']['name'])
+                context['ballot_link'] = 'https://whocanivotefor.co.uk/elections/{}'.format(self.object.ballot_id)
```

Comment:
I reckon moving these to `get_person`, `get_party` etc on the `Leaflet` model would be cleaner than having it in the view code, and make changing the logic easier in future, as the data model changes (imaging that this might be used in more than one place)

Reply:
Good shout, updated :+1: